mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+
+    Parameters
+    ----------
+    data : ndarray (n_epochs, n_times)
+        Signal time series.
+    sfreq : float
+        Sampling frequency in Hz.
+    l_phase_freq : float
+        Lower edge of phase modulating frequency.
+    h_phase_freq : float
+        Upper edge of phase modulating frequency.
+    l_amp_freq : float
+        Lower edge of amplitude modulated frequency.
+    h_amp_freq : float
+        Upper edge of amplitude modulated frequency.
+    bin_num : int
+        Number of phase bins. Default is 18. (20 degrees * 18 = 360)
+    method : str
+        Filter method used. Default is 'fft'.
+    n_jobs : int
+        Number of jobs used for parallelization. Default 1.
+    surrogates : bool
+        Return surrogates values for data specific to PAC,
+        the phase series and amplitude envelope are shuffled for every trial.
+        Default value is False.
+
+    Returns
+    -------
+    normalized_amplitude : ndarray, shape (n_epochs, bin_num)
+        The normalized amplitude values across each bin.
+    phase_bins : array, shape (number of bins + 1,)
+        Binned phase points. Used for plotting.
+
+    References
+    ----------
+    Measuring phase-amplitude coupling between neuronal oscillations
+    of different frequencies.
+    Tort ABL, Komorowski R, Eichenbaum H, Kopell N., (J Neurophysiol. 2010)
+    """
+
+    # obtained number of trials
+    n_epochs = len(data)
+    # time points in each epochs (assuming epochs of equal length)
+    n_times = data.size / n_epochs
+    # reshape epochs into single time series for filtering
+    filter_data = data.reshape((1, n_times * n_epochs))[0]
+
+    # Filter data into phase modulating freq and amplitude modulated freq.
+    filter_x_fp = band_pass_filter(filter_data, sfreq, l_phase_freq,
+                                   h_phase_freq, method=method, n_jobs=n_jobs)
+    filter_x_fa = band_pass_filter(filter_data, sfreq, l_amp_freq, h_amp_freq,
+                                   method=method, n_jobs=n_jobs)
+
+    # epoch data back again after filtering
+    time_slices = np.arange(0, data.size + n_times, n_times)
+    x_fa = np.zeros(data.shape)
+    x_fp = np.zeros(data.shape)
+    for ind in range(1, n_epochs + 1):
+        x_fa[ind - 1] = filter_x_fa[time_slices[ind - 1]: time_slices[ind]]
+        x_fp[ind - 1] = filter_x_fp[time_slices[ind - 1]: time_slices[ind]]
+
+    if len(x_fp) != len(x_fa) or x_fp.ndim != x_fa.ndim:
+        raise RuntimeError('Phase and amplitude filtered data does'
+                           ' not have same dimensions')
+
+    # Calculate phase series of phase modulating signal
+    phase_series_fp = np.angle(hilbert(x_fp)) + np.pi
+
+    # Calculate amplitude envelope of amplitude modulated signal
+    amp_envelope_fa = np.abs(hilbert(x_fa))
+
+    if surrogates is True:
+        phase_series_fp = shuffle_data(phase_series_fp)
+        amp_envelope_fa = shuffle_data(amp_envelope_fa)
```

Comment:
@dengemann @choldgraf I didn't implement the phase scrambling code snippet mentioned above. This is because of the way I use shuffle_data directly on the phases here.